docs/user/quickstart.rst

```diff
@@ -11,346 +11,639 @@ Learning by Example
+                httpd.serve_forever()
 
-    def max_body(limit):
+        Again this code uses wsgiref, but you you can also run the above example using
+        any WSGI server, such as uWSGI or Gunicorn. For example:
 
-        def hook(req, resp, resource, params):
-            length = req.content_length
-            if length is not None and length > limit:
-                msg = ('The size of the request is too large. The body must not '
-                       'exceed ' + str(limit) + ' bytes in length.')
+        .. code:: bash
 
-                raise falcon.HTTPPayloadTooLarge(
-                    title='Request body is too large', description=msg)
+            $ pip install requests gunicorn
+            $ gunicorn things:app
 
-        return hook
+        On Windows where Gunicorn and uWSGI don't work yet you can use Waitress server
 
+        .. code:: bash
 
-    class ThingsResource:
+            $ pip install requests waitress
+            $ waitress-serve --port=8000 things:app
 
-        def __init__(self, db):
-            self.db = db
-            self.logger = logging.getLogger('thingsapp.' + __name__)
 
-        def on_get(self, req, resp, user_id):
-            marker = req.get_param('marker') or ''
-            limit = req.get_param_as_int('limit') or 50
+        To test this example go to the another terminal and run:
 
-            try:
-                result = self.db.get_things(marker, limit)
-            except Exception as ex:
-                self.logger.error(ex)
+        .. code:: bash
 
-                description = ('Aliens have attacked our base! We will '
-                               'be back as soon as we fight them off. '
-                               'We appreciate your patience.')
+            $ http localhost:8000/1/things authorization:custom-token
 
-                raise falcon.HTTPServiceUnavailable(
-                    title='Service Outage',
-                    description=description,
-                    retry_after=30)
+        To visualize the application configuration the :ref:`inspect` can be used:
 
-            # NOTE: Normally you would use resp.media for this sort of thing;
-            # this example serves only to demonstrate how the context can be
-            # used to pass arbitrary values between middleware components,
-            # hooks, and resources.
-            resp.context.result = result
+        .. code:: bash
+
+            falcon-inspect-app things_advanced:app
+
+        This would print for this example application:
+
+        .. code::
+
+            Falcon App (WSGI)
+            • Routes:
+                ⇒ /{user_id}/things - ThingsResource:
+                   ├── GET - on_get
+                   └── POST - on_post
+            • Middleware (Middleware are independent):
+                → AuthMiddleware.process_request
+                  → RequireJSON.process_request
+                    → JSONTranslator.process_request
+
```

Comment:
Are these new lines around `├── Process route responder` intended? A question to @CaselIT I guess :slightly_smiling_face:

Reply:
yes, it looked better to my eyes with the newlines, but feel free to remove them :)
They are added here
https://github.com/falconry/falcon/blob/master/falcon/inspect.py#L683
https://github.com/falconry/falcon/blob/master/falcon/inspect.py#L688

Reply:
If this is by design, no need to change anything _right now_.
I'm still to experiment with that tool more in depth :) (But it looks awesome)